lib/auth/join_azure_test.go

```diff
@@ -73,13 +73,13 @@ func (m *mockAzureVMClient) Get(_ context.Context, resourceID string) (*azure.Vi
 	return vm, nil
 }
 
-func (m *mockAzureVMClient) GetByVMID(_ context.Context, resourceGroup, vmID string) (*azure.VirtualMachine, error) {
+func (m *mockAzureVMClient) GetByVMID(_ context.Context, vmID string) (*azure.VirtualMachine, error) {
 	for _, vm := range m.vms {
-		if vm.VMID == vmID && (resourceGroup == types.Wildcard || vm.ResourceGroup == resourceGroup) {
+		if vm.VMID == vmID {
 			return vm, nil
 		}
```

Comment:
Does this PR remove resource group check entirely or is it still enforced somewhere else and this was just for listing/searching purposes?

Reply:
I double checked and the resource group checks happen later based there: https://github.com/gravitational/teleport/pull/49968/files#diff-bef843e2704c1c08eea17403dca811318623ae14da002c1fb52e98b8a3c77f33R289

So this change does not affect how we check the joining instance RG.